src/Driver/ChromeDriver/Downloader.php

```diff
@@ -200,6 +227,29 @@ private function getFilePath(string $location, OperatingSystem $operatingSystem)
         return $location . DIRECTORY_SEPARATOR . $this->getFileName($operatingSystem);
     }
 
+    private function getTmpFilePath(string $location, OperatingSystem $operatingSystem): string
+    {
+        if ($operatingSystem->equals(OperatingSystem::WINDOWS())) {
+            return $location . DIRECTORY_SEPARATOR . self::BINARY_WINDOWS_JSON . '/' . $this->getFileName($operatingSystem);
+        }
+
+        if ($operatingSystem->equals(OperatingSystem::MACOS())) {
+            return implode(
+                DIRECTORY_SEPARATOR,
+                [$location, self::BINARY_MAC_JSON, $this->getFileName($operatingSystem)]
+            );
+        }
+
+        if ($operatingSystem->equals(OperatingSystem::LINUX())) {
+            return implode(
+                DIRECTORY_SEPARATOR,
+                [$location, self::BINARY_LINUX_JSON, $this->getFileName($operatingSystem)]
+            );
+        }
+
+        return $location . DIRECTORY_SEPARATOR . $this->getFileName($operatingSystem);
```

Comment:
Do you know why it fails? I don't see why it shouldn't work.

Reply:
Because of the / char i think

Reply:
Could you look a bit more into this? The full path that is being renamed already contains a `DIRECTORY_SEPARATOR`, so I don't see why adding more directory separators would suddenly break it. It would have a positive impact on the maintainability if we could solve this in the right place.